### PR TITLE
Fix NPEs

### PR DESCRIPTION
**Pull request by callunity**
[GLT-359]
* fix NPE in List Samples page
* use JsonSanitizer for List Libraries page
* add lastModifiers during bulk save Samples & Libraries
* more accurate error messages in bulk save Libraries
* remove identificationBarcode from bulk save Samples & Libraries, since it is currently not user-editable in the bulk table (which is to be replaced by Handsontable)

### CODE REVIEW

**Comment by apmasell** [comment]
:+1: LGTM

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 1]
```diff
@@ -487,8 +487,8 @@ public JSONObject bulkSaveLibraries(HttpSession session, JSONObject json) {
               library.setLibraryType(requestManager.getLibraryTypeByDescription(type));
               library.setLibrarySelectionType(requestManager.getLibrarySelectionTypeByName(selectionType));
               library.setLibraryStrategyType(requestManager.getLibraryStrategyTypeByName(strategyType));
-              library.setIdentificationBarcode(identificationBarcode);
-
+              library.setLastModifier(user);
+
```
Comment (frankie13) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: :space_invader: 
I think el formatador should trim whitespace?

**Comment by frankie13** [comment]
:+1: other than the horrifying whitespace violations :smile:

**Comment by callunity** [comment]
Fixed the spacing issue. :no_entry: :space_invader:

**Pull request merged**